coqpit/coqpit.py

```diff
@@ -608,56 +643,46 @@ def load_json(self, file_name: str) -> None:
         self = self.deserialize(dump_dict)  # pylint: disable=self-cls-assignment
         self.check_values()
 
-    def parse_args(self, args: argparse.Namespace) -> None:
+    def parse_args(self, args: Optional[argparse.Namespace] = None) -> None:
         """Update config values from argparse arguments with some meta-programming ✨.
 
         Args:
-            args (namespace): argeparse namespace as an output of ```argparse.parse_arguments()```.
+            args (namespace, optional): argeparse namespace as an output of ```argparse.parse_arguments()```. If unspecified will use a newly created parser with ```init_argparse()```.
 
         Returns:
             Coqpit: new config object with updated values.
         """
+        if not args:
+            parser = self.init_argparse()
+            args = parser.parse_args()
```

Comment:
This breaks when there is no argument for Coqpit but arguments for other parts and those arguments raise `undefined argument`